### PR TITLE
Don't re-export `clippy_utils::*`

### PR DESCRIPTION
**Pull request by camsteffen**
changelog: none

This should wrap up the utils-related really big changes. I personally don't see any more sub-modules to be added to `clippy_utils`.

### CODE REVIEW

**Comment by camsteffen** [comment]
Fun fact: With all theses utils changes, the number of multi-line utils imports went from 55 to 7!

File: clippy_lints/src/suspicious_trait_impl.rs [hunk 1]
```diff
@@ -159,7 +159,7 @@ fn check_binop(
     expected_ops: &[hir::BinOpKind],
 ) -> Option<&'static str> {
     let mut trait_ids = vec![];
-    let [krate, module] = crate::utils::paths::OPS_MODULE;
+    let [krate, module] = clippy_utils::paths::OPS_MODULE;
```
Comment (flip1995) [inline clippy_lints/src/suspicious_trait_impl.rs:162]:
```suggestion
    let [krate, module] = paths::OPS_MODULE;
```

File: clippy_lints/src/suspicious_trait_impl.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-use crate::utils::{get_trait_def_id, trait_ref_of_method};
 use clippy_utils::diagnostics::span_lint;
+use clippy_utils::{get_trait_def_id, trait_ref_of_method};
```
Comment (flip1995) [inline clippy_lints/src/suspicious_trait_impl.rs:2]:
```suggestion
use clippy_utils::{get_trait_def_id, paths, trait_ref_of_method};
```

**Comment by camsteffen** [comment]
Rebased and added path imports

**Comment by flip1995** [comment]
This will collide with pretty much every PR, but mostly with #6896. But I think rebasing everything else on this is easier, than rebasing this. So...

@bors r+ p=10

Thanks! (I really like the net -55 LoC)

**Comment by camsteffen** [comment]
> This will collide with pretty much every PR

Sorry everybody.

**Pull request merged**